src/auth0/handlers/organizations.js

```diff
@@ -0,0 +1,208 @@
+
+    const params = { id: org.id };
+    const { connections } = org;
+
+    delete org.connections;
+    delete org.name;
+    delete org.id;
+
+    await this.client.organizations.update(params, org);
+
+    const connectionsToRemove = existingConnections.filter(c => !connections.find(x => x.connection_id === c.connection_id));
+    const connectionsToAdd = connections.filter(c => !existingConnections.find(x => x.connection_id === c.connection_id));
+    const connectionsToUpdate = connections.filter(c => existingConnections.find(x => x.connection_id === c.connection_id && x.assign_membership_on_login !== c.assign_membership_on_login));
+
+    // Handle updates first
+    await Promise.all(connectionsToUpdate.map(conn => this.client.organizations
+      .updateEnabledConnection(Object.assign({ connection_id: conn.connection_id }, params), { assign_membership_on_login: conn.assign_membership_on_login })
+      .catch(() => {
+        throw new Error(`Problem updating Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    await Promise.all(connectionsToAdd.map(conn => this.client.organizations
+      .addEnabledConnection(params, _.omit(conn, 'connection'))
+      .catch(() => {
+        throw new Error(`Problem adding Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    await Promise.all(connectionsToRemove.map(conn => this.client.organizations
+      .removeEnabledConnection(Object.assign({ connection_id: conn.connection_id }, params))
+      .catch(() => {
+        throw new Error(`Problem removing Enabled Connection ${conn.connection_id} for organizations ${params.id}`);
+      })));
+
+    return params;
+  }
+
+  async updateOrganizations(updates, orgs) {
+    await this.client.pool.addEachTask({
+      data: updates || [],
+      generator: item => this.updateOrganization(item, orgs).then((data) => {
+        this.didUpdate(data);
+        this.updated += 1;
+      }).catch((err) => {
+        throw new Error(`Problem updating ${this.type} ${this.objString(item)}\n${err}`);
+      })
+    }).promise();
+  }
+
+  async getType() {
+    if (this.existing) {
+      return this.existing;
+    }
+
+    if (!this.client.organizations || typeof this.client.organizations.getAll !== 'function') {
+      return [];
+    }
+
+    try {
+      const organizations = await this.getAllOrganizations();
+      for (let index = 0; index < organizations.length; index++) {
+        const connections = await this.client.organizations.connections.get({ id: organizations[index].id });
+        organizations[index].connections = connections;
+      }
+      this.existing = organizations;
+      return this.existing;
+    } catch (err) {
+      if (err.statusCode === 404 || err.statusCode === 501) {
+        return [];
+      }
+      throw err;
+    }
+  }
+
+  async getAllOrganizations() {
```

Comment:
Instead of doing the pagination yourself, if you call `this.client.organizations.getAll({paginate: true})` then `client` should be able to do all the paging logic for you, and you'll get back a single array containing all of their data.

Reply:
checking